sklearn/calibration.py

```diff
@@ -622,7 +637,13 @@ def _fit_classifier_calibrator_pair(
         # Reshape binary output from `(n_samples,)` to `(n_samples, 1)`
         predictions = predictions.reshape(-1, 1)
 
-    sw_test = None if sample_weight is None else _safe_indexing(sample_weight, test)
+    if sample_weight is not None:
+        sw_test = _safe_indexing(sample_weight, test)
+        # Check that the sample_weight dtype is consistent with the predictions
+        # to avoid unintentional upcasts.
+        sw_test = _check_sample_weight(sw_test, predictions, dtype=predictions.dtype)
+    else:
+        sw_test = None
```

Comment:
Do we need this check here? The only place `_fit_classifier_calibrator_pair` is called is above where we ensure correct dtype of sample_weight just once.

Reply:
Unfortunately, I don't think it's possible to centralize the dtype check of sample weights as explained in https://github.com/scikit-learn/scikit-learn/pull/30873/files#r1967404296.